src/WebJobs.Script.WebHost/Diagnostics/MetricsEventManager.cs

```diff
@@ -53,7 +53,7 @@ public MetricsEventManager(IOptionsMonitor<AppServiceOptions> appServiceOptionsM
         /// <summary>
         /// Gets the collection of events that will be flushed on the next flush interval.
         /// </summary>
-        public ConcurrentDictionary<string, SystemMetricEvent> QueuedEvents { get; }
+        internal ConcurrentDictionary<EventKey, Lazy<SystemMetricEvent>> QueuedEvents { get; } = new ConcurrentDictionary<EventKey, Lazy<SystemMetricEvent>>();
```

Comment:
Curious: what does the `Lazy<>` get us here?

Reply:
When performing a `.GetOrAdd()` there's a race where you could generate many (there's no lock around the get/miss/add) and `Lazy<T>` is how to fix that - I'm actually missing the last commit here pushing the option into the `Lazt<T>` when forking this off, will fix!

Reply:
Oh that's why I left it off - `ExecutionAndPublication` is the default option, we should be good here: https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/Lazy.cs#L239